### PR TITLE
feat: migrate remix-netlify adapter

### LINKED ISSUE #82: Create a package for the Remix Netlify Functions adapter
Currently the template for Netlify Functions uses the `@remix-run/netlify` package. Bring in all that code into the netlify/remix-compute project and create a package with that same code that the template can use.

### PR DESCRIPTION
**Pull request by mcansh**
## Related Tickets & Documents

- Closes #82

## QA Instructions, Screenshots, Recordings

Note that for the time being, it's expected that the E2E tests fail for a PR from an external contributor. They don't get access to the token to run them. I've checked and they do pass when running manually.

CleanShot 2023-07-10 at 15 15 45

### For Serverless Demo

1. From the root of the project, run `npm run build:demo`
2. Run `ntl serve` to start the serverless demo site.
3. Run `npm run e2e` to start up Cypress.
4. Choose end to end tests from the Cypress UI.
5. Run the smoke test for the serverless demo site.
6. All tests pass.

### For Edge Demo

1. From the root of the project, run `npm run build:edge-demo`
2. Run `ntl serve` to start the serverless demo site.
3. Run `npm run e2e` to start up Cypress.
4. Choose end to end tests from the Cypress UI.
5. Run the smoke test for the serverless demo site.
6. All tests pass.

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/remix-compute/issues/new/choose) before writing your code 🧑‍💻. This
      ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a
      typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 📖. This ensures your code follows our style
      guide and passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**
image

### CODE REVIEW

**Comment by nickytonline** [comment]
Thanks for this @mcansh! I already have an E2E test suite for the Netlify Remix Edge Runtime that I can use for this adapter as well.

**Comment by mcansh** [comment]
> Thanks for this @mcansh! I already have an E2E test suite for the Netlify Remix Edge Runtime that I can use for this adapter as well.

sweet! i can bring over our [current tests](https://github.com/remix-run/remix/tree/main/packages/remix-netlify/__tests__) too if you'd like

**Comment by nickytonline** [comment]
> > Thanks for this @mcansh! I already have an E2E test suite for the Netlify Remix Edge Runtime that I can use for this adapter as well.
> 
> sweet! i can bring over our [current tests](https://github.com/remix-run/remix/tree/main/packages/remix-netlify/__tests__?rgh-link-date=2023-05-02T21%3A24%3A03Z) too if you'd like

Yeah, the unit tests would be great! Thanks!

**Comment by mcansh** [comment]
> The E2E tests are good, just trying to figure out why when I approve the run for an external contributor it still borks.

i think it was related to this being a "Draft PR" up until now.

File: package.json
Comment (nickytonline) [inline package.json:8]: naming (blocking): It's explicit, but we've discussed internally before about the name of this whenever we'd get around to it, and we landed on `@netlify/remix-adapter`

File: packages/remix-serverless-adapter/__tests__/binary-types.test.ts
Comment (nickytonline) [inline packages/remix-serverless-adapter/__tests__/binary-types.test.ts:1]: thanks for bringing in all the tests! 😍

File: packages/remix-adapter/package.json
Comment (MichaelDeBoey) [inline packages/remix-adapter/package.json:35]:
```suggestion
    "@remix-run/node": "1.17.0"
```

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by MichaelDeBoey** [comment]
@nickytonline Is there anything that still needs to be done before we can merge this one?

**Comment by nickytonline** [comment]
> @nickytonline Is there anything that still needs to be done before we can merge this one?

I've had a few things on the go. Will look to getting this reviewed.

File: packages/remix-adapter/package.json
Comment (mcansh) [reply packages/remix-adapter/package.json:35]: _probably_ better to do `"@remix-run/node": "^1.17.0"`

my local npm config at the time had exact pinned deps configured

**Comment by nickytonline** [comment]
It's normal that E2E tests are currently failing. External contributors don't have access to the token to run the GH action for E2E tests.

File: packages/remix-adapter/package.json
Comment (nickytonline) [reply packages/remix-adapter/package.json:35]: I've bumped it to latest, 1.18.1

**Comment by nickytonline** [comment]
@mcansh and @MichaelDeBoey, looking to get this merged next week. I've made a bunch of changes myself, so I can't approve it. I'll let someone on my team do that.

Thanks for the work on this both of you!

File: packages/remix-adapter/src/index.ts
Comment (MichaelDeBoey) [inline packages/remix-adapter/src/index.ts:1]: I think we can't forget to install the globals

```suggestion
import './globals'

export type { GetLoadContextFunction, RequestHandler } from './server'
```

`globals.ts`:
```ts
import { installGlobals } from '@remix-run/node'

installGlobals()
```

File: packages/demo-site/netlify.toml
Comment (MichaelDeBoey) [inline packages/demo-site/netlify.toml:1]: Do we want to have the `headers` part we have in the current template as well?

https://github.com/remix-run/remix/blob/8e2e1ab661dc2dcf803d4c45586f23c0973a3e5e/templates/netlify/netlify.toml#L10-L13

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: packages/demo-site/netlify.toml
Comment (nickytonline) [reply packages/demo-site/netlify.toml:2]: It's in the root package.json. This is just the location of the netlify.toml, but when it runs for this monorepo project, it runs in the context of the root of the project.

File: packages/demo-site/netlify.toml
Comment (nickytonline) [reply packages/demo-site/netlify.toml:1]: I'll take a peek at this, but probably yes. Maybe that wasn't a part of the template when @mcansh initially PRed this.

File: packages/demo-site/package.json
Comment (nickytonline) [reply packages/demo-site/package.json]: The E2E tests are for all the demo site, so yeah, n0t required in that project.

**Review by MichaelDeBoey — APPROVED** [review]

**Review by MarcL — DISMISSED** [review]
👍🏻 Looks good to me but I'm not a Remix expert!

**Review by MarcL — DISMISSED** [review]
:shipit:

**Review by MarcL — APPROVED** [review]

**Pull request merged**
